tests/processor/test_base.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import tempfile
+import os
+
+import pytest
+import mock
+
+from assayist.processor.base import Analyzer as BaseAnalyzer
+
+
+class DummyAnalyzer(BaseAnalyzer):
+    """A dummy analyzer to test non-static methods."""
+
+    def run(self):
+        """Do nothing."""
+        return
+
+
+@pytest.mark.parametrize('build_info,expected', [
+    ({'extra': None}, False),
+    ({'extra': {'something': []}}, False),
+    ({'extra': {'container_koji_task_id': 12345}}, True),
+])
+def test_is_container_build(build_info, expected):
+    """Test that the is_container_build method properly parses the passed-in build info."""
+    assert BaseAnalyzer.is_container_build(build_info) is expected
+
+
+def test_claim_container_file_whole_dir():
```

Comment:
I wonder if we even care about claiming directories. If all we were left with at the end of analysis was a tree of otherwise-empty directories, I think we'd say there's nothing left to do.